cache/cache.go

```diff
@@ -57,3 +60,34 @@ func (c *Cache) WriteTo(w io.Writer) (int64, error) {
 	}
 	return total, nil
 }
+
+// Index generates an index for the given cache on a particular
+// column.  This is required for caches beyond a libnss-cache defined
+// size in order for them to be read correctly.
+func (c *Cache) Index(col int) bytes.Buffer {
+	var ordered []string
```

Comment:
You could use `make` here in order to reduce the allocations which could happen in the loop.

Reply:
Indeed.  I have made this change.